### PR TITLE
remove redundant judgments during the HNSW search process

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: src/algorithm/hnswlib/hnswalg.cpp [hunk 1]
```diff
@@ -1404,13 +1404,9 @@ HierarchicalNSW::searchKnn(const void* query_data,
     }
 
     MaxHeap top_candidates;
-    if (num_deleted_) {
-        top_candidates =
-            searchBaseLayerST<true, true>(currObj, query_data, std::max(ef, k), isIdAllowed);
```
Comment (jiaweizone) [inline src/algorithm/hnswlib/hnswalg.cpp:1409]: Why remove this logic ?

Comment (inabao) [reply src/algorithm/hnswlib/hnswalg.cpp:1409]: num_deleted_ is the original logic for marking deletions in HNSW, but we no longer rely on this logic.

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by wxyucs — APPROVED** [review]
lgtm

**Pull request merged**